lib/mp-readline/readline.c

```diff
@@ -98,6 +99,26 @@ typedef struct _readline_t {
 
 STATIC readline_t rl;
 
+#if MICROPY_REPL_EMACS_WORDS_MOVE
+STATIC size_t cursor_count_word(int forward) {
+    const char *line_buf = vstr_str(rl.line);
+    size_t pos = rl.cursor_pos;
+    bool in_word = false;
+
+    while ((forward || 0 < pos) && (!forward || pos < vstr_len(rl.line))) {
+        if (unichar_isalnum(line_buf[pos + (forward - 1)])) {
+            in_word = true;
+        } else if (in_word) {
+            break;
+        }
+
+        pos += forward ?: forward - 1;
```

Comment:
This doesn't look right?

Reply:
Yes, `forward` is indeed used as kind of a boolean here, but the 1/0 representation allows producing smaller code, -8 bytes on stm32 this way (which is disappointing, I was hoping it'd be optimized correctly).

It's legit to add `-1` to a `size_t`. As long as it doesn't underflow... (Which can't happen here due to the loop checks). On ARM, it boils down to `add	r4, r8` with `r8=0` or `r8=0xffffffff`.

Reply:
Actually I was also talking about the `?:`. I didn't know it, turns out it's a GNU extension and so won't compile with MSVC and possibly other compilers so you should just write it out.

Reply:
Uh, you're right, I keep forgetting MP compiles w/ MSVC as well. It's indeed a GNU extension. I'll change.